src/test/java/echoServer/AcceptanceTest.java

```diff
@@ -0,0 +1,53 @@
+package echoServer;
+
+import org.junit.jupiter.api.Test;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.util.concurrent.TimeUnit;
+
+import static jdk.jfr.Timespan.SECONDS;
+
+public class AcceptanceTest {
+//    Create a process
+//      Create an instance of ProcessBuilder with a list of commands and arguments
+//      invoke the start() method on that instance
+//    Write to the process's input stream and read from its output stream?
+
+//Sets this process builder's standard input source. Subprocesses subsequently started by this object's start() method obtain their standard input from this source.
+//    If the source is Redirect.PIPE (the initial value), then the standard input of a subprocess can be written to using the output stream returned by Process.getOutputStream().
+
+    @Test
+    void itWorks() throws IOException, InterruptedException {
+        Process buildAndRunEchoServer = new ProcessBuilder("gradle", "build", "&&", "gradle", "run").start();
```

Comment:
**issue**: Let's not shell out to Gradle within our tests.

If we want to write our acceptance test in this fashion, let's call `Main.main()` instead. It is simpler and it avoids a circular dependency between Gradle and our application code.

**Current**: We have a loop/cycle in our dependency graph.

```mermaid
flowchart
    App("Application (java/main)")
    Tests("Tests (java/test)")
    Gradle --> Tests
    Gradle --> App
    Tests --> App
    Tests --> Gradle
```

**Desired**: The dependency graph is acyclic

```mermaid
flowchart
    App("Application (java/main)")
    Tests("Tests (java/test)")
    Gradle --> Tests
    Gradle --> App
    Tests --> App
```

Alternatively, if we want to test through the shell, maybe we should use a shell script to test instead of Java.

```mermaid
flowchart
    AT("Acceptance Test (shell script)")
    App("Application (java/main)")
    Tests("Tests (java/test)")
    AT --> Gradle
    Gradle --> Tests
    Gradle --> App
    Tests --> App
```